[suhayl13] iP

suhayl13 opened the pull request:
# Lexi

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
## Lexi frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. Download from [here](https://github.com/suhayl13/ip/releases/tag/v1.0).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😺

AND it's **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events

If you are a java programmer, you can use it to practice too. Here is the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

suhayl13 committed:
Rename main class

suhayl13 committed:
Add greeting

suhayl13 committed:
Add support for echo

suhayl13 committed:
Add support for list

suhayl13 committed:
Add support for mark as done

suhayl13 committed:
Add support for three types of tasks

suhayl13 committed:
Run tests for current functionalities

suhayl13 committed:
Create LexiException for error handling

suhayl13 committed:
Add support for error handling

suhayl13 committed:
Add support for deleting tasks

suhayl13 committed:
Use enums

suhayl13 committed:
Add data.txt for saving data of users

suhayl13 committed:
Move enum as a separate file

suhayl13 committed:
Add getter method for retrieving task name

suhayl13 committed:
Add getter methods for deadline and event class

suhayl13 committed:
Put enum into a different file

suhayl13 committed:
Add support for save

suhayl13 committed:
Merge branch 'branch-Level-7'
Implement feature for saving the data

suhayl13 committed:
Implement dates and times feature

suhayl13 committed:
Merge branch 'branch-Level-8'
Implement feature for date and time compatibility

suhayl13 committed:
Add new classes

suhayl13 committed:
Add new classes

suhayl13 committed:
Add support for bye command

suhayl13 committed:
Merge branch 'branch-A-MoreOOP'

suhayl13 committed:
Update .gitignore

suhayl13 committed:
Put classes into packages

suhayl13 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

suhayl13 committed:
Modify build.gradle file

suhayl13 committed:
Refactor storage and main class

suhayl13 committed:
Refactor storage class

suhayl13 committed:
Add support for junit test

suhayl13 committed:
Merge branch 'branch-A-JUnit'

suhayl13 committed:
Modify build.gradle

suhayl13 committed:
Fix bug for MarkCommand

suhayl13 committed:
Refactor lexi and ui classes

suhayl13 committed:
Add javadocs to all public classes and methods

suhayl13 committed:
Follow coding standard for all classes

suhayl13 committed:
Add support for find function

suhayl13 committed:
Merge branch 'branch-A-JavaDoc'

suhayl13 committed:
Merge branch 'branch-A-CodingStandard'

suhayl13 committed:
Merge branch 'branch-Level-9'

SeanWong2 commented on src/main/java/lexi/Lexi.java:
```diff
@@ -0,0 +1,71 @@
+     * This method starts the main loop that reads commands from the user, processes them,
+     * and executes the corresponding actions until the user decides to exit.
+     */
+    public void run() {
```
I like the how you abstracted the functions into other classes

SeanWong2 commented on src/main/java/lexi/parser/Parser.java:
```diff
@@ -0,0 +1,215 @@
+        String[] parts = response.split(" ");
+        try {
+            Commands command = Commands.valueOf(parts[0].toUpperCase());
+            switch (command) {
```
I like the naming of the variables and functions, it makes it easy to understand

SeanWong2 commented on src/main/java/lexi/task/DatedTask.java:
```diff
@@ -0,0 +1,29 @@
+package lexi.task;
```
I like the use of a date time class and camel casing

SeanWong2 commented on src/main/java/lexi/storage/Storage.java:
```diff
@@ -0,0 +1,145 @@
+package lexi.storage;
```
I like the constant use of PascalCase for classes and camelCase for variables throughout the code

ckaayy commented on src/main/java/lexi/parser/Parser.java:
```diff
@@ -0,0 +1,215 @@
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
```
I like how you separated the imports into various logical blocks!

ckaayy commented on src/main/java/lexi/storage/Storage.java:
```diff
@@ -0,0 +1,145 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.Scanner;
```
Should the imports be in lexicographical order?

ckaayy commented on src/test/java/lexi/task/TodoTest.java:
```diff
@@ -0,0 +1,49 @@
+package lexi.task;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Should the statements follow a consistent format? There is a line separator for your other files. (Just a nit)
```suggestion
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;
```

ckaayy reviewed (commented):
LGTM! :). 
Just a few nits

suhayl13 committed:
Implement GUI for Lexi

suhayl13 committed:
Merge branch 'branch-Level-10'

SeanWong2 commented on src/main/java/lexi/command/AddCommand.java:
```diff
@@ -0,0 +1,57 @@
+        this.task = task;
+    }
+
+    /**
```
I like the detailed description of your methods

SeanWong2 commented on src/main/java/lexi/Lexi.java:
```diff
@@ -0,0 +1,55 @@
+     * Constructs a new Lexi application instance.
+     * Initializes the UI, loads tasks from the specified file, and prepares the task list.
+     */
+    public Lexi() {
```
I like the abstraction of your methods into different classes like storage and command

SeanWong2 commented on src/main/java/lexi/ui/DialogBox.java:
```diff
@@ -0,0 +1,76 @@
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
```
I like the alphabetical order of your imports

suhayl13 committed:
Add checkstyle to code

suhayl13 committed:
Merge branch 'branch-A-CheckStyle'

suhayl13 committed:
Ensure compliance with checkstyle for test classes

SeanWong2 reviewed (commented):
Looks good in general

suhayl13 committed:
Add assertions to most of classes

The classes lacked assertions. So assertions were added
so that unintended values occuring at specific points in the methods.

Adding assertions would help identify potential bugs in the code,
making debugging and testing easier.

suhayl13 committed:
Merge pull request #2 from suhayl13/branch-A-Assertions

Add assertions

suhayl13 committed:
Refactor code for better readability

suhayl13 committed:
Merge branch 'master' into branch-A-CodeQuality

suhayl13 committed:
Merge pull request #3 from suhayl13/branch-A-CodeQuality

Refactor code for better readability

suhayl13 committed:
Add stream to Parser

suhayl13 committed:
Merge branch 'master' into branch-A-Streams

suhayl13 committed:
Merge pull request #4 from suhayl13/branch-A-Streams

Add streams to parser

suhayl13 committed:
Change order of assignment in Lexi:getResponse

The program return a NullPointerException when list command
is executed when the data file is empty as ListCommand:execute
throws a LexiException when the list of tasks are empty.

As a result, the commandType assignment does not execute as
the program exits the code and reaches the catch block. This
later causes NullPointerException in DialogBox:changeDialogStyle
as commandType is null. It is null as the program exited the
try block before the assignment statement.

Let's perform the assignment of the commandType before executing
the command itself.

suhayl13 committed:
Add support for CI

suhayl13 committed:
Test CI

suhayl13 committed:
Add support for update

suhayl13 committed:
Merge branch 'branch-C-Update'
Add support for update feature

Currently, if users want to update a specific task, they need to delete that task
and add a new task that matches their updated version.

This causes inconvenience for the users who merely want to update a task.

Let's
* Add an Update Command
* Handle the logic of replacing the old task with the updated one
* Update the storage data accordingly

suhayl13 committed:
Remove I/O test from Github Workflows

suhayl13 committed:
Improve GUI by rounding corners of display picture in dialog box

suhayl13 committed:
Add title to chatbox

suhayl13 committed:
Add Ui.png

suhayl13 committed:
Update readme.md to include user guide

suhayl13 committed:
Use AI to declutter deeply nested code in Storage class
